src/Validation/Validator/HashmapValidator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ScrumWorks\OpenApiSchema\Validation\Validator;
+
+use ScrumWorks\OpenApiSchema\Validation\BreadCrumbPath;
+use ScrumWorks\OpenApiSchema\Validation\Result\ValidationResultBuilderFactoryInterface;
+use ScrumWorks\OpenApiSchema\Validation\Result\ValidationResultBuilderInterface;
+use ScrumWorks\OpenApiSchema\Validation\ValueValidator;
+use ScrumWorks\OpenApiSchema\ValueSchema\HashmapSchema;
+
+final class HashmapValidator extends AbstractValidator
+{
+    private HashmapSchema $schema;
+
+    private ValueValidator $valueValidator;
+
+    public function __construct(
+        ValidationResultBuilderFactoryInterface $validationResultBuilderFactory,
+        HashmapSchema $schema,
+        ValueValidator $valueValidator
+    ) {
+        parent::__construct($validationResultBuilderFactory, $schema);
+
+        $this->schema = $schema;
+        $this->valueValidator = $valueValidator;
+    }
+
+    protected function doValidation(
+        ValidationResultBuilderInterface $resultBuilder,
+        $data,
+        BreadCrumbPath $breadCrumbPath
+    ): void {
+        if (! $this->validateNullable($resultBuilder, $data, $breadCrumbPath)) {
+            return;
+        }
+
+        if (! \is_array($data)) {
```

Comment:
mozna jsem to tu nekde prehledl, ale kontroluje se, ze klice jsou stringy? - protoze je to jedina moznost pro hashmap vznikle z JSON

Reply:
Mimojine to klade druhou otazku - pokud by jsme meli
```
/**
 * @var array<int, string>
 */
property array $hashmap;
```
tak ja to samozrejme pro potreby hashmap muzu prevest na `array<string, string>` ale tim nam utece info o tom klici... otakza je jak vyresit dilema ze z JSON nam dojde vzdy string... jedine co me napada to dat jako nejake additional info pro hashmap, ze je klic int

Reply:
Jestli by nahodou nebylo reseni nepouzivat json_decode bez assoc na true a hash mapa by validovala stdclass

Reply:
urcite je to taky moznost, otazka je jestli ti to neprinese praci navic - protoze pak nebudes mit genericky typ array, ale nejsem proti